java-squid/src/main/java/org/sonar/java/resolve/Symbols.java

```diff
@@ -123,6 +124,17 @@ public String toString() {
     unknownType.supertype = null;
     unknownType.interfaces = ImmutableList.of();
     unknownSymbol.type = unknownType;
+
+    anyType = new JavaType.ClassJavaType(unknownSymbol) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this a named "static" inner class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2694)